### ISSUE TITLE
D7 Upgrade: undefined function entity_load() in user_load_multiple()

### ISSUE THREAD

**Issue by laryn** [backdrop/backdrop-issues#5549]
## Description of the bug

I spun up a fresh Drupal 7 site, installed one module that I'm working on and then tried the upgrade process. I ran into this error unexpectedly while trying to run `update.php`:

```
Error: Call to undefined function entity_load() in user_load_multiple() (line 266 of /app/web/core/modules/user/user.module)
```

It looks like `entity` is enabled (status = 1) in the system table at this point, with a schema of 0.

**Comment by laryn** [comment]
The module I'm working on has nothing to do with users, by the way (Conditional Fields).

**Comment by laryn** [comment]
Just to keep my current project moving forward, I've hacked the line above to use `require_once` to load `user.module` and that works for me for now. The upgrade completes at this point, although another strange warning:

> Warning: Use of undefined constant VIEWS_STORAGE_DEFAULT - assumed 'VIEWS_STORAGE_DEFAULT' (this will throw an Error in a future version of PHP) in filter_update_1004() (line 491 of /app/web/core/modules/filter/filter.install).

**Comment by argiepiano** [comment]
> 



> In this case the D7 site did not have Entity API enabled.

OK. Then this is something I haven't seen before. TBH the D7 upgrade process is getting so little attention and seems to be so fraught with problems, yet it's such an important aspect; the future of Backdrop may very well rely on a smooth experience upgrading from D7

**Comment by laryn** [comment]
Yes, I agree we should make sure the red carpet is rolled out for D7 upgrades...

**Comment by argiepiano** [comment]
@laryn, I'm interested in looking further into this. So far I've been unable to reproduce this error. I'm wondering if the Backdrop codebase is a clean install? And are you starting up from a Standard D7 profile of the latest version? What D7 modules did you have enabled?

**Comment by argiepiano** [comment]
Also, can you clarify when you see this message? In my testing `user_load_multiple()` is never called during the upgrade process of a clean Standard D7 site - it's only called when you try to log in after the update.

**Comment by laryn** [comment]
@argiepiano I believe it was just a vanilla D7 with conditional fields installed (and one conditional field defined in a content type). The error occurred during the database updates step.

**Comment by argiepiano** [comment]
@laryn, I've done additional testing, and am still unable to reproduce this error.

1. In my D7 clean install, I enabled Conditional Fields, created a new content type, added two fields (one of them boolean) and created a dependency so that field B was shown if the boolean was set to true. All works well.
2. I created content.
3. I followed the usual upgrade steps with a Backdrop codebase that included Conditional Fields in the modules folder
4. The upgraded proceeded smoothly, no errors. I logged in the upgraded site. The content type and content were both migrated, and the Conditional Field works well

Any special settings you used in the D7 Conditional Field relationship?

**Comment by laryn** [comment]
@argiepiano 

I had nothing fancy configured on the D7 site, just as little as I could configure to test the CF update process. I can replicate it locally.

I took the `backdrop-ready.sql` database (compressed version attached below) and did the following:

1. Install a vanilla version of Backdrop in a lando container
2. Download Conditional Fields into the modules directory
3. Use `lando drush sql-cli < ./backdrop-ready.sql` to import the D7 database
4. Make sure `update_free_access` is set to `TRUE` in `settings.php`
5. Run `core/update.php`

It goes through a bunch of updates but then crashes here:
![Screen Shot 2022-03-21 at 12 57 29 PM](https://user-images.githubusercontent.com/871421/159335508-8097603b-6070-41dc-ae25-e423a05f6a97.jpg)

[backdrop-ready.sql.zip](https://github.com/backdrop/backdrop-issues/files/8318159/backdrop-ready.sql.zip)

**Comment by olafgrabienski** [comment]
@laryn you seem to install Backdrop before the upgrade, and @argiepiano you do not, correct? (The docs recommend the latter, see https://docs.backdropcms.org/documentation/step-3-upgrade-the-drupal-site.) Could that make a difference here?

**Comment by argiepiano** [comment]
@laryn, I tested an upgrade with the database dump you attached. I did not find any issue. I'm starting to suspect that the problem is with your setup. I'm using a manual clean, local install of Backdrop (I use MAMP for database and web server). I don't have/use lando, so I can't test that. I have the dev version of Backdrop's Conditional Fields. Since I don't have the admin credentials for the dump you attached, I could not log in, but the upgrade did not through any errors, and I could see the site fine. Here are screenshots of the BAckdrop site after upgrade. 

<img width="809" alt="Screen Shot 2022-03-21 at 7 22 08 PM" src="https://user-images.githubusercontent.com/9938978/159416554-978d0d1a-559b-407b-b7f9-c1d81af5dded.png">
<img width="714" alt="Screen Shot 2022-03-21 at 7 22 21 PM" src="https://user-images.githubusercontent.com/9938978/159416563-70bf5cd1-c6bc-4a23-ae40-9502e5066fa6.png">

**Comment by argiepiano** [comment]
@olafgrabienski comment may be on to something. I'm not familiar with the drush command `sql-cli <`, but I would make sure that you "wipe out" the old Backdrop database before importing the new (D7) one.

**Comment by laryn** [comment]
I did try using `drush sql-drop` before the import command, with the same results. I'll try to test in a different (maybe non-Lando) setup another time. Not likely today though.

**Comment by laryn** [comment]
I suppose it's possible that having config files from a vanilla install still in the config folder could be part of the issue.
